examples/example-userspace/examples/p0f.rs

```diff
@@ -0,0 +1,49 @@
+// for details about the p0f, please refer to https://github.com/p0f/p0f
+
+use futures::stream::StreamExt;
+use probes::p0f::TcpSignature;
+use redbpf::{load::Loader, xdp};
+
+fn probe_code() -> &'static [u8] {
+    include_bytes!(concat!(env!("OUT_DIR"), "/target/bpf/programs/p0f/p0f.elf"))
+}
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> std::result::Result<(), String> {
+    let xdp_mode = xdp::Flags::DrvMode;
+    let interfaces: Vec<String> = vec!["eth0".to_string()];
+
+    let mut loaded = Loader::load(probe_code()).map_err(|err| format!("{:?}", err))?;
+
+    for interface in &interfaces {
+        println!("Attach p0f on interface: {} with mode {:?}", interface, xdp_mode);
+        for prog in loaded.xdps_mut() {
+            prog.attach_xdp(interface, xdp_mode)
+                .map_err(|err| format!("{:?}", err))?;
+        }
+    }
+
+    let _ = tokio::spawn(async move {
+        while let Some((name, events)) = loaded.events.next().await {
+            for event in events {
+                match name.as_str() {
+                    "tcp_signatures" => {
+                        let tcp_sig =
+                            unsafe { std::ptr::read(event.as_ptr() as *const TcpSignature) };
```

Comment:
I think `ptr::read_unaligned` is proper here since the result of `event.as_ptr()` is not guaranteed to be aligned at the `mem::align_of::<TcpSignature>()`

Reply:
@rhdxmr sorry for delayed reply and thanks a lot for your review, I've fixed this and added the "signed-off-by". if you got available time, could you please help to check whether it is possible to simplify it and increase the tcp option count, which is controlled by 

```rust
for _ in 0..p0f::MAX_TCP_OPT - 6 {
```

it is 5 at most and it would be nice to support more

Reply:
If the loop iteration count is increased, it is failed to load this BPF program with this BPF verifier log..
```
BPF program is too large. Processed 1000001 insn
processed 1000001 insns (limit 1000000) max_states_per_insn 23 total_states 88465 peak_states 1892 mark_read 18
```
The BPF bytecode of p0f only contains about 2k instructions, so the 1M limit problem is not directly caused by the number of instructions generated by compiler. The thing is the number of executed instructions at runtime.

So I think this problem can not be solved by bounded-loop or BPF-to-BPF call that just reduce the size of BPF bytecode.


This link suggests to use tail call.
https://stackoverflow.com/a/70848246/12851147
But I am not sure.

Although this might solve the problem, a big modification should be involved. So I think it would be better to deal the improvement by another PR if it is possible.